File: dynabuffers-python/dynabuffers/ast/structural/UnionOptions.py
class UnionOptionsOptions:
    def __init__(self, primary: bool, deprecated: bool):
        self.primary = primary
        self.deprecated = deprecated


class UnionOptions:

    def __init__(self, options: UnionOptionsOptions):
        self.options = options

    def is_primary(self):
        return self.options.primary

    def is_deprecated(self):
        return self.options.deprecated

File: dynabuffers-java/src/main/kotlin/dynabuffers/ast/structural/UnionOptions.kt
package dynabuffers.ast.structural

import dynabuffers.api.IType

class UnionOptions(private val options: UnionOptionsOptions) : IType {

    data class UnionOptionsOptions(val primary:Boolean, val deprecated: Boolean, val implicit: Boolean)

    fun isPrimary() = options.primary
    fun isDeprecated() = options.deprecated
    fun isImplicit() = options.implicit

}
